### ISSUE TITLE
Communication breaks with nats-py 2.1.3 and .NET clients (Headers Issue)

### ISSUE THREAD

**Issue by pohlt** [nats-io/nats.py#322]
When we upgraded from nats-py 2.1.2 to 2.1.3 (Python 3.10.5), a .NET client using https://www.nuget.org/packages/NATS.Client
complained about an unexpected header structure and disconnected from the NATS broker.

Some context:
* We actually do set our own headers - nothing fancy, just strings of course - *BUT we do have two headers with empty strings.*
* We use Core NATS, so no JetStream.
* We saw the issue when sending **Request**s from nats-py to a .NET client. We didn't test Reply/Pub/Sub.

Reverting back to 2.1.2 solved the issue as expected.

**Comment by wallyqs** [comment]
Thanks @pohlt looks like there was a regression in the Python client in latest release with the published headers:

```sh
HMSG foo 1 107 125
NATS/1.0timestamp: 2022-06-15T19:08:14.639020
type: rpc
command: publish_state
trace_id: 
span_id: 

Hello from Python!
```

but should have been the following with a `\r\n`, just noticed that this also causes when sending headers from Python client to the Go client.

```
HMSG foo 1 109 127
NATS/1.0
timestamp: 2022-06-15T19:08:14.639020
type: rpc
command: publish_state
trace_id: 
span_id: 

Hello from Python!
```

**Comment by kung-foo** [comment]
We have the opposite issue. Messages with headers sent from the Go client, show up in 2.1.3 with no headers. Downgrading to 2.1.2 fixed the issue and brought back the headers.

**Comment by woeye** [comment]
@kung-foo I can confirm. We have the same issue here: I am setting headers in Go:

    msg := &nats.Msg{}
    msg.Header = make(nats.Header)
    msg.Header.Set("status", fmt.Sprint(status))

However if I receive this message in Python then msg.headers is always set to None (headers=None)

Will try 2.1.2 next.

**Comment by wallyqs** [comment]
Fixed via #328

**Comment by pohlt** [comment]
Just wanted to confirm that everything works with 2.1.4 when communicating with Go and C# clients.

Thanks a lot! 🙏🏼
